compiler/dyno/lib/resolution/scope-queries.cpp

```diff
@@ -349,7 +355,8 @@ static bool doLookupInImports(Context* context,
         // find it in the contents
         const Scope* symScope = is.scope();
         LookupConfig newConfig = LOOKUP_DECLS |
-                                 LOOKUP_IMPORT_AND_USE;
+                                 LOOKUP_IMPORT_AND_USE |
+                                 LOOKUP_SKIP_PRIVATE_VIS;
```

Comment:
Why does skipPrivateVisibilities get reset here? Doesn't this mean we'll go through one private use / import (we didn't `continue` on line 348 above), but then get stuck at the next one? Do we want it to be conditionally set (only if `skipPrivateVisibilities` is true?)

Reply:
This is the case where we are going to follow through a `use`/`import` to recursively continue scope-resolution lookup. We want to skip private `use`s "from here on out" on the call stack, because we have now left the original module we were searching in.

It isn't reset so much as set -- this is the only place where we ever set the `LOOKUP_SKIP_PRIVATE_VIS` flag, and from here on down the call stack it's propagated if present.

The reason we won't go through one private use/import and then get stuck on the next is because we are not changing `skipPrivateVisibilities` at all between loop iterations; we are setting the flag for use further down the call stack.